normalize.go

```diff
@@ -21,8 +21,16 @@ import (
 	"go.uber.org/atomic"
 )
 
+var _ fmt.Formatter = (*redactFormatter)(nil)
+
 // RedactLogEnabled defines whether the arguments of Error need to be redacted.
-var RedactLogEnabled atomic.Bool
+var RedactLogEnabled atomic.String
+
+const (
+	RedactLogEnable  string = "ON"
+	RedactLogDisable        = "OFF"
+	RedactLogMarker         = "MARKER"
```

Comment:
It's more extensible to have this be a function.

Reply:
What do you mean? Could you explain it a little bit?

Reply:
`RedactLogEnabled` could be maintained as `atomic.Bool`. The redaction function could then be a configurable setting. Currently the library user can only select `RedactLogMarker`. Which is equivalent to the function adding brackets. Previously there was a function that replaced with a question mark. With a function, these changes could be made by configuring library usage rather than changing code in the library.

Reply:
> Which is equivalent to the function adding brackets.

This is intended. Since there are lots of usage of `errors.FastGenWithArgs....`, it is just too verbose to modify every invokation to use `brackets`. My goal is to add `brackets` or replacing with questionmark automatically without even noticing.

> `atomic.Bool`

I agreed that probably it will be good if it is `atomic.Bool` with an extra flag. But the thing is I need some atomicity guarentees to prevent data races in TiDB. It is better just one variable.

What do you think a struct wrapper around `atomic.String`? Probably with `RedactLogEnabled.SetOn/SetOff/SetMarker`?

Reply:
BTW, this is set by `tidb_redact_log`.

Reply:
In the real world it is already a problem that humans ideally would be alerted about if there are multiple sessions making conflicting configuration changes at the same time. Making these atomic doesn't seem to solve any such issues.

I see how this is a problem for test cases. It seems like these are still global variables that will cause problems for test cases though? It seems test cases will need to lock these settings in some way. But if these were not globals but instead settings that can be passed around then this would not be an issue.

Loggers have this issue. They usually support changing a global default but also defining a scope in which settings are valid- such a scope would make testing easy.

Reply:
>  Making these atomic doesn't seem to solve any such issues.

It is just a guard around cpu data races.

>  It seems like these are still global variables 

Yes, but there is no cpu races which will be captured by tests and panic.

>  It seems test cases will need to lock these settings in some way. But if these were not globals but instead settings that can be passed around then this would not be an issue.

That is ideal. But it is out of the scope of this feature. We have this problem for years. There are lots, lots of such global variables. Some are even "required" because the bad API between two components.

I hope someone can solve this eventually.